### DISCUSSION TITLE
Managing OnChange with multiple arrays

### DISCUSSION THREAD

**Discussion by Electacle** [iddan/react-spreadsheet#187] [Q&A]
Hello, I use several data tables with this react module, like this:

data = [ [...arrays], [...arrays...] ]

then I use a useState on this data, then a loop to launch the different spreadsheets 

 <Spreadsheet key={index} className='table' columnLabels={stringLabelCol} rowLabels={stringLabelRow} data={stateData[index]} onChange={setData} />

Only, it seems that the OnChange doesn't work when it comes to handling multiple tables and I don't know what modification I need to make so that it can handle this situation. 

If someone could enlighten me, I would be grateful!
Thanks for reading!

**Comment by dpejic** [comment]
Hello. You have to put only one array of data in spreadsheet component. So when you change data in spreadsheet component you have to change only that array index in state.
